dev/release-build.sh

```diff
@@ -0,0 +1,324 @@
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+function exit_with_usage {
+  cat << EOF
+
+release-build - Creates build distributions from a git commit hash or from HEAD.
+
+SYNOPSIS
+
+usage: release-build.sh [--release-prepare | --release-publish | --release-snapshot]
+
+release-build.sh --release-prepare --releaseVersion="2.0.0" --developmentVersion="2.1.0-SNAPSHOT" [--releaseRc=rc1] [--tag="v2.0.0"] [--gitCommitHash="a874b73"] [--dryRun]
+release-build.sh --release-publish --gitCommitHash="a874b73"
+release-build.sh --release-publish [--gitCommitHash="a874b73"]
+
+DESCRIPTION
+
+Use maven infrastructure to create a project release package and publish
+to staging release location (https://dist.apache.org/repos/dist/dev/bahir)
+and maven staging release repository.
+
+--release-prepare --releaseVersion="2.0.0" --developmentVersion="2.1.0-SNAPSHOT" [--releaseRc="rc1"] [--tag="v2.0.0"] [--gitCommitHash="a874b73"]
+This form execute maven release:prepare and upload the release candidate distribution
+to the staging release location.
+
+--release-publish --gitCommitHash="a874b73"
+Publish the maven artifacts of a release to the Apache staging maven repository.
+Note that this will publish both Scala 2.10 and 2.11 artifacts.
+
+--release-snapshot: [--gitCommitHash="a874b73"]
+Publish the maven snapshot artifacts to Apache snapshots maven repository
+Note that this will publish both Scala 2.10 and 2.11 artifacts.
+
+
+OPTIONS
+
+--releaseVersion     - Release identifier used when publishing
+--developmentVersion - Release identifier used for next development cyce
+--releaseRc          - Release RC identifier used when publishing, default 'rc1'
+--tag                - Release Tag identifier used when taging the release, default 'v$releaseVersion'
+--gitCommitHash      - Release tag or commit to build from, default master HEAD
+--dryRun             - Dry run only, mostly used for testing.
+
+A GPG passphrase is expected as an environment variable
+
+GPG_PASSPHRASE - Passphrase for GPG key used to sign release
+
+
+EXAMPLES
+
+release-build.sh --release-prepare --releaseVersion="2.0.0" --developmentVersion="2.1.0-SNAPSHOT" [--releaseRc="rc1"] [--tag="v2.0.0"] [--gitCommitHash="a874b73"]
+release-build.sh --release-publish --gitCommitHash="a874b73"
+release-build.sh --release-publish [--gitCommitHash="a874b73"]
+
+
```

Comment:
Very nice. I would remove the examples from under SYNOPSIS since you have them under EXAMPLES.

Your last example, should that read `release-snapshot: ` as opposed to `release-publish ` ?
```console
release-build.sh --release-prepare --releaseVersion="2.0.0" --developmentVersion="2.1.0-SNAPSHOT" [--releaseRc="rc1"] [--tag="v2.0.0"] [--gitCommitHash="a874b73"]
release-build.sh --release-publish --gitCommitHash="a874b73"
release-build.sh --release-publish [--gitCommitHash="a874b73"]
```
-->
```console
release-build.sh --release-prepare --releaseVersion="2.0.0" --developmentVersion="2.1.0-SNAPSHOT" [--releaseRc="rc1"] [--tag="v2.0.0"] [--gitCommitHash="a874b73"]
release-build.sh --release-publish --gitCommitHash="a874b73"
release-build.sh --release-snapshot: [--gitCommitHash="a874b73"]
```